source/server/admin/admin.h

```diff
@@ -244,6 +246,31 @@ class AdminImpl : public Admin,
     TimeSource& time_source_;
   };
 
+  /**
+   * Implementation of OverloadManager that is never overloaded. Using this instead of the real
+   * OverloadManager keeps the admin interface accessible even when the proxy is overloaded.
+   */
+  struct NullOverloadManager : public OverloadManager {
+    struct NullThreadOverloadState : public ThreadLocalOverloadState {
+      NullThreadOverloadState(Event::Dispatcher& dispatcher) : dispatcher_(dispatcher) {}
+      const OverloadActionState& getState(const std::string&) override { return inactive_; }
+
+      const OverloadActionState inactive_ = OverloadActionState::Inactive;
+      Event::Dispatcher& dispatcher_;
+    };
+
+    NullOverloadManager(Event::Dispatcher& dispatcher) : thread_local_overload_state_(dispatcher) {}
+    void start() override {}
+    ThreadLocalOverloadState& getThreadLocalOverloadState() override {
+      return thread_local_overload_state_;
+    }
+    bool registerForAction(const std::string&, Event::Dispatcher&, OverloadActionCb) override {
+      return false;
+    }
+
+    NullThreadOverloadState thread_local_overload_state_;
```

Comment:
I don't think this object is a thread local.  I'm not sure what's the right way to resolve this issue.  ThreadLocalOverloadState only has const methods in its interface so it is safe for the null overload manager to be a singleton instead of a thread local.  I wonder if renaming to OverloadManagerState would help, or further add to the confusion.

Reply:
Made this object explicitly thread-local instead of relying on const accessors.